File: 657-RobotReturnToOrigin/javascript/solution.js
// Last updated: 3/6/2026, 12:45:31 AM
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function (moves) {
    let x = 0;
    let y = 0;
    for (let i of moves) {
        if (i == "U") {
            y++;
        }
        else if (i == "D") {
            y--;
        }
        else if (i == "L") {
            x++;
        }
        else if (i == "R") {
            x--;
        }
    }
    return x == 0 && y == 0;
};

File: 657-RobotReturnToOrigin/cpp/solution.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        int x = 0;
        int y = 0;
        for (int i : moves) {
            if (i == 'U') {
                y++;
            } else if (i == 'D') {
                y--;
            } else if (i == 'L') {
                x++;
            } else if (i == 'R') {
                x--;
            }
        }
        return x == 0 && y == 0;
    }
};